### ISSUE TITLE
AlertDialog add support for `allowsKeyboardConfirmation` prop

### ISSUE THREAD

**Issue by ktabors** [adobe/react-spectrum#855]
# 🙋 Feature Request

AlertDialog is missing the feature `allowsKeyboardConfirmation` as defined in the API at https://github.com/adobe/react-spectrum/blob/main/packages/%40react-types/dialog/src/index.d.ts#L83.

## 🤔 Expected Behavior

`allowsKeyboardConfirmation` - Pressing the `enter` key closes the AlertDialog as if primary action button was clicked, which would call `onPrimaryAction()`. If the user has focused one of the dialog buttons or a text-area that component/element should handle the `enter` key event, which in the case of the button, its `usePress` hook will handle that.

## 😯 Current Behavior

Not implemented.
You can see existing behavior AlertDialog behavior at https://react-spectrum.adobe.com/react-spectrum/AlertDialog.html

## 💁 Possible Solution

For similar functionality, the `escape` key closing the AlertDialog is handled by this code in an aria hook https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/overlays/src/useOverlay.ts#L69-L75

## 🧢 Your Company/Team
RSP

**Comment by intergalacticspacehighway** [comment]
@ktabors I just read the documentation, we already have autoFocusButton which can be set to primary (which would trigger the enter functionality). Is there any additional use case for using allowsKeyboardConfirmation?

**Comment by ktabors** [comment]
The use case I understand it is this allows someone to close the dialog with confirm regardless of where the focus is.

**Comment by dannify** [comment]
^ with the added caveat that pressing `enter` while focused on a component that already handles it, behaves as expected. E.g. if focus is on the Cancel button, pressing `enter` will cancel the dialog as normal.

**Comment by intergalacticspacehighway** [comment]
@dannify Yeah. When I read the feature I felt the easiest way to implement it would be to move the focus to the primary button when allowsKeyboardConfirmation is used but we already have a similar prop.

**Comment by dannify** [comment]
Right yeah I get that. This option is more for the situation where it is not the first focus event of the dialog. Maybe there are some form elements in a dialog such as a checkbox. In that case, hitting Enter would allow the user to "confirm" the dialog without having to keyboard focus the confirm button first.

**Comment by luisorbaiceta** [comment]
Is this still needed? I can work on it

**Comment by devongovett** [comment]
We decided not to support this prop. As an alternative, you can use `autoFocusButton="primary". When the user presses enter, this will trigger the primary action (unless they tab somewhere else). This has the benefit of showing a focus ring as well.
